### PR TITLE
Python 3 compatibility fixes

### PR DESCRIPTION
**Pull request by PetrDlouhy**
This prevents OL from usage in Python 3 only environment.

### CODE REVIEW

**Comment by ahocevar** [comment]
Will this still work with Python 2?

**Comment by PetrDlouhy** [comment]
It should, although I tested it only on Python 2.7

**Comment by fredj** [comment]
It seems to work in python 2.6:
```
$ python2.6 
Python 2.6.8 (unknown, Jan 26 2013, 14:35:25) 
[GCC 4.7.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> print('ok')
ok
>>> 

```

**Comment by PetrDlouhy** [comment]
@fredj The print with parentheses is not the only change I made.

I made some more testing now. As I was testing this, I bumped into more incompatibilities with Python 3, so I added more fixes.

I tried to build with all compressors on Python 2.6 and it works OK, except uglify_js which didn't work on 2.6 before.

Although Python 2.5 doesn't support "except Exception as e" syntax.
The question is, if we can drop Python 2.5 support, because writing code compatible with 2.5-3 versions is quite complicated: http://stackoverflow.com/questions/11285313/try-except-as-error-in-python-2-5-python-3-x
Python 2.6 was released in 2008, so it is still quite old release and can is backward compatible with Python 2.5, which is quite antique. And anyway, uglify_js needs 2.7 with current code.

**Comment by fredj** [comment]
I think we can support python >= 2.7

**Comment by ahocevar** [comment]
> I think we can support python >= 2.7

Agreed.

File: lib/OpenLayers/Format/WFST/v1_0_0.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ OpenLayers.Format.WFST.v1_0_0 = OpenLayers.Class(
                 var prefix = options.featurePrefix;
                 var node = this.createElementNSPlus("wfs:Query", {
                     attributes: {
-                        typeName: (prefix ? prefix + ":" : "") +
+                        typeName: (options.featureNS ? prefix + ":" : "") +
```
Comment (tschaub) [inline lib/OpenLayers/Format/WFST/v1_0_0.js]: These look like changes that are unrelated to your Python 3 work.  It is better if you create a dedicated branch for your pull request instead of using `master`.

File: lib/OpenLayers/Format/WFST/v2_0_0.js [hunk 2]
```diff
@@ -170,7 +170,7 @@ OpenLayers.Format.WFST.v2_0_0 = OpenLayers.Class(
                 var prefix = options.featurePrefix;
                 var node = this.createElementNSPlus("wfs:Query", {
                     attributes: {
-                        typeNames: (prefix ? prefix + ":" : "") +
+                        typeNames: (options.featureNS ? prefix + ":" : "") +
```
Comment (tschaub) [inline lib/OpenLayers/Format/WFST/v2_0_0.js]: This is also an unrelated change.

File: lib/OpenLayers/Layer/Google.js [hunk 3]
```diff
@@ -471,4 +471,4 @@ OpenLayers.Layer.Google = OpenLayers.Class(
  * Property: OpenLayers.Layer.Google.cache
  * {Object} Cache for elements that should only be created once per map.
  */
-OpenLayers.Layer.Google.cache = {};
\ No newline at end of file
+OpenLayers.Layer.Google.cache = {};
```
Comment (tschaub) [inline lib/OpenLayers/Layer/Google.js]: More unrelated changes.

File: tests/Format/WFST/v1_0_0.html [hunk 4]
```diff
@@ -75,7 +75,7 @@
             featureType: "states",
             featurePrefix: "topp"
         });
-        var exp = "topp:states";
+        var exp = "states";
```
Comment (tschaub) [inline tests/Format/WFST/v1_0_0.html]: Also not Python.

File: tests/Format/WFST/v2_0_0.html [hunk 5]
```diff
@@ -278,7 +278,7 @@
             featureType: "states",
             featurePrefix: "topp"
         });
-        var exp = "topp:states";
+        var exp = "states";
```
Comment (tschaub) [inline tests/Format/WFST/v2_0_0.html]: No Python here either.

**Comment by PetrDlouhy** [comment]
@tschaub Thanks for your comments to the code. The unrelated changes was caused bad imported commit from bad rebase. I have merged all Python 3 changes into single commit.

I have added slight additional change - usage of OrderedDicts instead of simple dicts which ensures stability of output JS file. This is essential during testing to ensure if the output is correct.

I have tested all the compressors with Python 2.7 and 3.4

**Comment by PetrDlouhy** [comment]
@ahocevar Can you please pull this? Or is there anything I should adjust?

**Comment by ahocevar** [comment]
I hope this doesn't break things for anyone, but I agree it's time to move away from deprecated Python versions.

**Comment by PetrDlouhy** [comment]
@ahocevar Thanks.

**Comment by ahocevar** [comment]
@PetrDlouhy It looks like we do not have a [CLA](https://github.com/openlayers/cla) from you. Can you sign one please?

**Comment by PetrDlouhy** [comment]
@ahocevar OK, I signed.
